### PR TITLE
feat(desktop): MCP server management UI in settings (KAT-2387)

### PR DESCRIPTION
**Pull request by gannonh**
## Summary
- add a main-process MCP config bridge with typed CRUD, redaction, provenance, and atomic write/readback behavior
- add Settings → MCP UI with guided stdio/HTTP bearer editor, row actions, and provenance/warning surfaces
- wire live MCP refresh/reconnect status inspection and available-tools reporting through IPC/preload into renderer state
- add deterministic Electron e2e MCP settings coverage and S03 live smoke documentation
- expand MCP bridge/service unit coverage to satisfy desktop coverage thresholds in pre-push validation

## Validation
- `cd apps/desktop && npx vitest run src/main/__tests__/mcp-config-bridge.test.ts src/main/__tests__/mcp-service.test.ts src/main/__tests__/mcp-ipc.test.ts src/renderer/components/settings/__tests__/McpServerPanel.test.tsx`
- `cd apps/desktop && bun run typecheck`
- `cd apps/desktop && npx playwright test e2e/tests/mcp-settings.e2e.ts`
- `cd apps/desktop && KATA_SYMPHONY_BIN_PATH= bun run test`

## Scope notes
- shared global `~/.kata-cli/agent/mcp.json` management only (project overlay remains read-only provenance warning)
- non-OAuth stdio/http-bearer entries only; no full OAuth onboarding claims

## Summary by CodeRabbit

* **New Features**
  * Added MCP server management interface in Settings with a new "MCP" tab.
  * Users can create, configure, edit, and delete MCP server entries.
  * Real-time server connectivity status monitoring with tool discovery display.
  * Support for both stdio and HTTP-based server configurations.
  * Bearer token authentication for HTTP servers with environment variable support.

* **Tests**
  * Added comprehensive test coverage for MCP settings, configuration, and server connectivity.

<h3>Greptile Summary</h3>

This PR introduces the complete MCP server management feature for Kata Desktop: a config bridge for `~/.kata-cli/agent/mcp.json` CRUD with atomic writes and redaction, an editor dialog, a settings panel, a status-refresh/reconnect service, IPC wiring, and unit + E2E coverage. Two defects need attention before merge:

- **Bearer token erasure on edit** (`McpServerEditorDialog.tsx` line 170): editing an existing HTTP server and saving without entering a new token silently deletes the stored credential. `toServerInput` passes `bearerToken: ''` (not `undefined`), causing `normalizeHttpServerForWrite` to `delete nextHttpServer.bearerToken` on every blank-field save — directly contradicting the UI hint that says \"Existing inline token remains unless replaced.\" Fix: `bearerToken: draft.bearerToken || undefined`.
- **HTTP `notifications/initialized` breaks on 204** (`mcp-service.ts` lines 192-196): `postJsonRpc` throws on any non-2xx status and calls `response.json()` unconditionally; a spec-compliant server returning `204 No Content` to a notification will be misreported as unreachable.

<h3>Confidence Score: 4/5</h3>

Not safe to merge as-is: bearer token data loss and HTTP 204 handling break real-world workflows

Two P1 defects with clear reproduction paths: (1) users editing an HTTP server and leaving the bearer token blank silently lose their stored credentials — the UI's own hint contradicts the actual behaviour; (2) spec-compliant MCP HTTP servers returning 204 to notifications/initialized will be false-reported as unreachable. Both are one-to-three line fixes with no architectural risk.

McpServerEditorDialog.tsx (line 170) and mcp-service.ts (lines 192-196)

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| apps/desktop/src/renderer/components/settings/McpServerEditorDialog.tsx | Root of bearer-token data-loss: toServerInput passes bearerToken:'' instead of undefined on edit, causing normalizeHttpServerForWrite to delete the stored credential |
| apps/desktop/src/main/mcp-service.ts | inspectHttpServer awaits postJsonRpc for notifications/initialized; spec-compliant servers returning 204 No Content will throw, misreporting healthy servers as unreachable |
| apps/desktop/src/main/mcp-config-bridge.ts | Atomic temp-file write/readback and redaction logic are correct; normalizeHttpServerForWrite correctly handles undefined but is exposed to the editor passing empty string |
| apps/desktop/src/main/ipc.ts | MCP IPC handlers (list, get, save, delete, refresh, reconnect) registered and cleaned up correctly; no issues found |
| apps/desktop/src/preload/index.ts | MCP preload methods correctly forward all six IPC channels to the renderer via contextBridge |
| apps/desktop/src/shared/types.ts | McpServerInput, McpHttpServerInput, status types, and IPC_CHANNELS are well-defined; bearerToken correctly typed as optional |
| apps/desktop/src/renderer/atoms/mcp.ts | Jotai write atoms for load, save, delete, and status actions correctly manage loading/error/success state |
| apps/desktop/src/renderer/components/settings/McpServerPanel.tsx | Panel correctly wires load, save, delete, refresh, and reconnect; provenance warning and error alerts render correctly |
| apps/desktop/src/renderer/components/settings/McpServerRow.tsx | Row component correctly renders server info, status badge, and two-step delete confirmation; no issues found |
| apps/desktop/src/renderer/components/settings/SettingsPanel.tsx | MCP tab correctly wired alongside existing provider/symphony/general/appearance tabs |
| apps/desktop/src/main/__tests__/mcp-config-bridge.test.ts | Good CRUD, redaction, and provenance coverage; missing a test for save-with-empty-bearerToken to catch the editor bug |
| apps/desktop/src/main/__tests__/mcp-ipc.test.ts | IPC routing tests correctly verify handler delegation to config bridge and service stubs |
| apps/desktop/src/main/__tests__/mcp-service.test.ts | Service tests cover refresh and reconnect paths; 204 No Content edge case for HTTP notifications not tested |
| apps/desktop/src/renderer/components/settings/__tests__/McpServerPanel.test.tsx | Helper function tests cover provenance labels, server summarization, status labels, and draft creation |
| apps/desktop/e2e/tests/mcp-settings.e2e.ts | E2E covers stdio add/edit/delete, reconnect failure, and malformed config; HTTP bearer token round-trip not covered |
| apps/desktop/e2e/fixtures/electron.fixture.ts | mcpConfigPath fixture correctly isolates test config in a temp dir via KATA_DESKTOP_MCP_CONFIG_PATH env var |
| apps/desktop/e2e/fixtures/mcp-stdio-server.mjs | Minimal stdio MCP fixture server correctly implements initialize and tools/list with framed JSON-RPC |
| apps/desktop/e2e/tests/app-shell.e2e.ts | Minimal addition of MCP tab navigation assertion to existing shell test; no issues |
| apps/desktop/docs/uat/M005/S03-MCP-SMOKE.md | UAT smoke checklist document; no code issues |

</details>

</details>

<h3>Sequence Diagram</h3>

```mermaid
sequenceDiagram
    participant U as User (Edit Dialog)
    participant D as McpServerEditorDialog
    participant B as McpConfigBridge
    participant F as mcp.json

    U->>D: Opens edit for HTTP server (stored bearerToken)
    Note over D: draft.bearerToken = '' (not loaded for security)
    U->>D: Leaves bearer field blank, clicks Save
    D->>D: toServerInput() → bearerToken: '' (not undefined)
    D->>B: saveServer({ bearerToken: '' })
    B->>B: normalizeHttpServerForWrite()<br/>'' !== undefined → enters block<br/>''.trim() falsy → delete bearerToken
    B->>F: Writes config WITHOUT bearerToken
    Note over F: Stored credential lost!

    Note over D,B: Fix: bearerToken: draft.bearerToken || undefined<br/>→ undefined skips the delete block → token preserved
```

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
This is a comment left during a code review.
Path: apps/desktop/src/renderer/components/settings/McpServerEditorDialog.tsx
Line: 163-185

Comment:
**Bearer token silently erased on edit**

When editing an existing HTTP server that has a stored inline bearer token, `draft.bearerToken` is initialised to `''` (the token is never loaded back into the form). `toServerInput` passes `bearerToken: draft.bearerToken`, so the saved input always carries `bearerToken: ''` — not `undefined`.

In `normalizeHttpServerForWrite` (`mcp-config-bridge.ts` lines 623-629), the guard `if (input.bearerToken !== undefined)` is `true` for `''`, so the code enters the block, finds `''.trim()` falsy, and runs `delete nextHttpServer.bearerToken` — erasing the stored credential on every save where the user leaves the field blank. The UI hint at line 416 — "Existing inline token remains unless replaced." — describes the intended behaviour, not what actually happens.

```suggestion
  if (draft.transport === 'http') {
    return {
      name: draft.name.trim(),
      transport: 'http',
      enabled: draft.enabled,
      url: draft.url.trim(),
      auth: draft.auth,
      bearerToken: draft.bearerToken || undefined,
      bearerTokenEnv: draft.bearerTokenEnv,
    }
  }
```

How can I resolve this? If you propose a fix, please make it concise.

---

This is a comment left during a code review.
Path: apps/desktop/src/main/mcp-service.ts
Line: 192-196

Comment:
**HTTP `notifications/initialized` crashes on spec-compliant 204 responses**

`postJsonRpc` is called for a JSON-RPC notification. Per JSON-RPC 2.0 and the MCP HTTP transport spec, servers MUST NOT reply to notifications, or may reply with `204 No Content`. But `postJsonRpc` (line 544) throws on `!response.ok` (204 is not OK per the Fetch API), and line 548 calls `response.json()` unconditionally, which throws on an empty body. A spec-compliant MCP HTTP server will be misreported as unreachable despite being healthy.

Fire-and-forget the notification instead:
```ts
// notifications/initialized: fire-and-forget per JSON-RPC 2.0
void fetch(server.url, {
  method: 'POST',
  headers,
  body: JSON.stringify({
    jsonrpc: '2.0',
    method: 'notifications/initialized',
    params: {},
  }),
  signal: AbortSignal.timeout(this.requestTimeoutMs),
}).catch(() => { /* ignore — notification responses are not required */ })
```

How can I resolve this? If you propose a fix, please make it concise.
`````

</details>

<sub>Reviews (1): Last reviewed commit: ["test(desktop): harden MCP bridge/service..."](https://github.com/gannonh/kata/commit/229174b754606eff9f43c6d95fbb85b6de3051b4) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=27485037)</sub>

> Greptile also left **2 inline comments** on this PR.

### CODE REVIEW

File: apps/desktop/src/main/mcp-service.ts [hunk 1]
```diff
@@ -0,0 +1,629 @@
...
+          formatJsonRpcError('initialize', initializeResponse.error),
+          checkedAt,
+        )
+      }
+
+      await postJsonRpc(server.url, {
+        jsonrpc: '2.0',
+        method: 'notifications/initialized',
+        params: {},
+      }, headers, this.requestTimeoutMs)
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/main/mcp-service.ts]: Addressed in commit 34a26a56.

HTTP `notifications/initialized` now uses fire-and-forget notification handling (`postJsonRpcNotification`) that does not parse response bodies, so 204/empty responses are accepted. Added coverage in `mcp-service.test.ts` for a 204 notification response path.
